fuel-core/src/schema/dap.rs

```diff
@@ -192,4 +192,272 @@ impl DapMutation {
+                return Ok(self::gql_types::RunResult {
+                    state: self::gql_types::RunState::Completed,
+                    breakpoint: None,
+                })
+            }
+            // The transaction was already completed earlier, so it cannot be resumed
+            Err(err) => return Err(async_graphql::Error::new(format!("VM error: {err:?}"))),
+        };
+
+        let dbgref = state.debug_ref();
+
+        Ok(self::gql_types::RunResult {
+            state: match dbgref {
+                Some(_) => self::gql_types::RunState::Breakpoint,
+                None => self::gql_types::RunState::Completed,
+            },
+            breakpoint: dbgref.and_then(|d| match d {
+                DebugEval::Continue => None,
+                DebugEval::Breakpoint(bp) => Some(bp.into()),
+            }),
+        })
+    }
+}
+
+mod gql_types {
+    //! GraphQL type wrappers
+    use async_graphql::*;
+
+    use fuel_types::Word;
+    #[cfg(feature = "debug")]
+    use fuel_vm::prelude::Breakpoint as FuelBreakpoint;
+
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, InputObject)]
+    pub struct ContractId {
+        value: [u8; 32],
+    }
+    impl From<&fuel_types::ContractId> for ContractId {
+        fn from(id: &fuel_types::ContractId) -> Self {
+            Self { value: **id }
+        }
+    }
+    impl From<fuel_types::ContractId> for ContractId {
+        fn from(id: fuel_types::ContractId) -> Self {
+            Self { value: *id }
+        }
+    }
+    impl From<ContractId> for fuel_types::ContractId {
+        fn from(id: ContractId) -> Self {
+            Self::from(id.value)
+        }
+    }
+
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, SimpleObject)]
+    pub struct OutputContractId {
+        value: [u8; 32],
+    }
+    impl From<&fuel_types::ContractId> for OutputContractId {
+        fn from(id: &fuel_types::ContractId) -> Self {
+            Self { value: **id }
+        }
+    }
+    impl From<fuel_types::ContractId> for OutputContractId {
+        fn from(id: fuel_types::ContractId) -> Self {
+            Self { value: *id }
+        }
+    }
+    impl From<OutputContractId> for fuel_types::ContractId {
+        fn from(id: OutputContractId) -> Self {
+            Self::from(id.value)
+        }
+    }
+
+    #[derive(Debug, Clone, Copy, InputObject)]
+    pub struct Breakpoint {
+        contract: [u8; 32],
+        pc: Word,
+    }
+
+    #[cfg(feature = "debug")]
+    impl From<&FuelBreakpoint> for Breakpoint {
+        fn from(bp: &FuelBreakpoint) -> Self {
+            Self {
+                contract: **bp.contract(),
+                pc: bp.pc(),
+            }
+        }
+    }
+
+    #[cfg(feature = "debug")]
+    impl From<Breakpoint> for FuelBreakpoint {
+        fn from(bp: Breakpoint) -> Self {
+            Self::new(bp.contract.into(), bp.pc)
+        }
+    }
+
+    #[derive(Debug, Clone, Copy, SimpleObject)]
+    pub struct OutputBreakpoint {
```

Comment:
Why should an output breakpoint differ from a regular breakpoint? Consider adding a comment to clarify.

Reply:
Only because the a single type cannot act as both input and output type in `async-graphql`, or at least I couldn't figure out how it could.  I'll add a comment to clarify that.

Reply:
Nevermind, updating to `async-graphql` version `3.0` seems to allow this.

Reply:
Is there an issue to track this upgrade?

Reply:
https://github.com/FuelLabs/fuel-core/issues/310